apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -102,42 +155,45 @@ def prepare(element_html, data):
     pl.check_answers_names(data, answer_name)
 
     check_indentation = pl.get_boolean_attrib(element, "indentation", INDENTION_DEFAULT)
-    grading_method = pl.get_string_attrib(
-        element, "grading-method", GRADING_METHOD_DEFAULT
+    grading_method = pl.get_enum_attrib(
+        element, "grading-method", GradingMethodType, GRADING_METHOD_DEFAULT
+    )
+    feedback_type = pl.get_enum_attrib(
+        element, "feedback", FeedbackType, FEEDBACK_DEFAULT
     )
-    feedback_type = pl.get_string_attrib(element, "feedback", FEEDBACK_DEFAULT)
 
-    if grading_method in ["dag", "ranking"]:
+    if (
+        grading_method is GradingMethodType.DAG
+        or grading_method is GradingMethodType.RANKING
+    ):
         partial_credit_type = pl.get_string_attrib(element, "partial-credit", "lcs")
```

Comment:
The partial credit type should be an enum as well (and only be retrieved once).

Reply:
done